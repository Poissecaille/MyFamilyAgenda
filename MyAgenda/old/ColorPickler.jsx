import React, {useEffect, useState} from 'react';
import {Button, View} from 'react-native';
import ColorPicker from 'react-native-wheel-color-picker';

export const ColorPickerObject = props => {
  const [currentColor, setCurrentColor] = useState('#fffff');
  return (
    <>
      {props.visible && (
        <View>
          <ColorPicker
            ref={r => {}}
            color={currentColor}
            onColorChangeComplete={color => {
              setCurrentColor(color);
              props.colorChangeHandler(color);
            }}
            thumbSize={40}
            row={false}
            swatchesLast={false}
            sliderHidden={true}
          />
          {/* <Button title="btn-picker" onPress={() => this.picker.revert()} /> */}
        </View>
      )}
    </>
  );
};
