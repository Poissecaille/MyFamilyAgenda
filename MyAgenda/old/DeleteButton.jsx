import React from 'react';
import styled from 'styled-components/native';
import { colors } from '../Colors';
import { RegularText } from '../RegularText';

const ButtonStyle = styled.TouchableOpacity`
  position:absolute;
  align-items: center;
  background-color: ${colors.LightRed};
  width: 20%;
  height:30%;
  left:180px;
  top:30px;
  border-radius: 10px;
  z-index:1
`;

export const DeleteButton = props => {
    return (
        <ButtonStyle onPress={props.onPress}>
            <RegularText value={props.title}></RegularText>
        </ButtonStyle>
    );
};
