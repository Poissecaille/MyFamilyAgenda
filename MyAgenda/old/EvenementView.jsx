import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

export const EvenementView = (props) => {

  return (
    <>
      <View style={styles.centeredView}>
        <Text style={styles.title}> {props.name} (R.S) </Text>
        <Text style={styles.regular}> {props.date} à {props.place} </Text>
        <Text style={styles.regular}> {props.description} </Text>
        <Text style={styles.regular}> Invités : {props.guests}</Text>
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  centeredView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  title: {
    fontSize: 40,
    fontWeight: 'bold',
  },
  regular: {
    fontSize: 25,
    fontWeight: 'lighter',
  }
});