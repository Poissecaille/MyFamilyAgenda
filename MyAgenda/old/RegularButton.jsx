import React from 'react';
import styled from 'styled-components/native';
import { colors } from '../src/utility/Global';
import { RegularText } from '../src/components/RegularText';


const ButtonStyle = styled.TouchableOpacity`
  align-items: center;
  background-color: ${colors.Darker1};
  width: 80%;
  padding: 20px;
  margin-top: 10px;
  margin-left: 30px;
  border-radius: 20px;
`;

// const styles = StyleSheet.create({
//   button: {
//     backgroundColor: colors.Darker1,
//     textAlign: 'center',
//     width: '80%',
//     padding: 20,
//     marginTop: 10,
//     marginLeft: 30,
//     borderRadius: 20
//   }
// })

export const RegularButton = props => {
  return (
    <ButtonStyle idTest={"onPress"} onPress={props.onPress}>
      <RegularText idTest={"text"} value={props.title}></RegularText>
    </ButtonStyle>
    // <TouchableOpacity style={styles.button}>
    //   <RegularText value={props.title}></RegularText>
    // </TouchableOpacity>
  );
};
