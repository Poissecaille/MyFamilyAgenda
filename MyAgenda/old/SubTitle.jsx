import React from 'react';
import styled from "styled-components/native";
import { colors } from "../src/utility/Global";


const SubTitleStyle = styled.Text`
    font-size: 15px;
    text-align: left;
    font-family: PlayfairDisplay-SemiBold;
    color: ${colors.Clearer1}
`;

export const SubTitle = (props) => {
    return (
        <SubTitleStyle
            style={props.style}
            idTest="subtitle"
        >
            {props.value}
        </SubTitleStyle>
    )
}
