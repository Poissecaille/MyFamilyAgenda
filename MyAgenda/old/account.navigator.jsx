import React from 'react';
import { createStackNavigator } from '@react-navigation/stack';

import { AccountScreen } from '../src/components/Account';
import { LoginScreen } from '../src/components/Login';
import { RegisterScreen } from '../src/components/Register';


const Stack = createStackNavigator();

export const AccountNavigator = () => (
  <Stack.Navigator headerMode="none">
    <Stack.Screen name="Main" component={AccountScreen} />
    <Stack.Screen name="Login" component={LoginScreen} />
    <Stack.Screen name="Register" component={RegisterScreen} />
  </Stack.Navigator>
);
