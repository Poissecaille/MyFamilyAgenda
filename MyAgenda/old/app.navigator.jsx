import React from 'react';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import {Ionicons} from '@expo/vector-icons';
import {Home} from '../../components/Home';
import {SettingsNavigator} from './settings.navigator';

const Tab = createBottomTabNavigator();

const TAB_ICON = {
  Restaurants: 'md-restaurant',
  Map: 'md-map',
  Settings: 'md-settings',
};

const createScreenOptions = ({route}) => {
  const iconName = TAB_ICON[route.name];
  return {
    tabBarIcon: ({size, color}) => (
      <Ionicons name={iconName} size={size} color={color} />
    ),
  };
};

export const AppNavigator = () => (
  <Tab.Navigator
  screenOptions={createScreenOptions}
  tabBarOptions={{
    activeTintColor: 'tomato',
    inactiveTintColor: 'gray',
  }}>
   <Tab.Screen name="Home" component={Home} />

   <Tab.Screen name="Settings" component={SettingsNavigator} />
 </Tab.Navigator>
);
