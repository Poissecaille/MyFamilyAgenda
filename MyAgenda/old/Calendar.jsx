import React from 'react';
import { useState } from 'react';
import { Calendar } from 'react-native-calendars';
import { LocaleConfig } from 'react-native-calendars';
import { ColorPickerObject } from './ColorPickler';
import { RegularButton } from './RegularButton';
import { colors } from '../src/utility/Global';

export const CalendarObject = () => {
  const [dateString, setDateString] = useState(
    new Date().toISOString().split('T')[0],
  ); // date formatted as 'YYYY-MM-DD' string
  const [picker, setPicker] = useState(false);
  const [color, setColor] = useState(colors.Primary_Color);
  const [markedDate, setMarkedDate] = useState('{}');

  LocaleConfig.locales['fr'] = {
    monthNames: [
      'Janvier',
      'Février',
      'Mars',
      'Avril',
      'Mai',
      'Juin',
      'Juillet',
      'Août',
      'Septembre',
      'Octobre',
      'Novembre',
      'Décembre',
    ],
    monthNamesShort: [
      'Janv.',
      'Févr.',
      'Mars',
      'Avril',
      'Mai',
      'Juin',
      'Juil.',
      'Août',
      'Sept.',
      'Oct.',
      'Nov.',
      'Déc.',
    ],
    dayNames: [
      'Dimanche',
      'Lundi',
      'Mardi',
      'Mercredi',
      'Jeudi',
      'Vendredi',
      'Samedi',
    ],
    dayNamesShort: ['Dim.', 'Lun.', 'Mar.', 'Mer.', 'Jeu.', 'Ven.', 'Sam.'],
    today: "Aujourd'hui",
  };
  LocaleConfig.defaultLocale = 'fr';

  const colorPickerHandler = () => {
    setPicker(!picker);
  };
  const colorChangeHandler = color => {
    setColor(color);
  };
  const updateMarkedDayHandler = date => {
    let tmp;
    if (typeof markedDate == 'string') {
      tmp = JSON.parse(markedDate);
    } else {
      tmp = markedDate;
    }
    if (!(date.dateString in tmp)) {
      tmp[date.dateString] = {
        selected: true,
        selectedColor: color,
        dotColor: color,
      };
    } else {
      tmp[date.dateString] = {
        selected: !tmp[date.dateString].selected,
        selectedColor: color,
        dotColor: color,
      };
    }
    setMarkedDate(JSON.stringify(tmp));
  };

  return (
    <>
      <RegularButton title={'Color picker'} onPress={colorPickerHandler} />
      <Calendar
        style={{
          marginTop: 80,
          padding: 20,
        }}
        // Initially visible month
        initialDate={dateString}
        // Minimum date that can be selected, dates before minDate will be grayed out.
        minDate={dateString}
        // Do not show days of other months in month page. Default = false
        hideExtraDays={true}
        onDayPress={date => {
          //   console.log('selected date', date, color);
          updateMarkedDayHandler(date);
        }}
        // Enable the option to swipe between months. Default = false
        enableSwipeMonths={true}
        markedDates={JSON.parse(markedDate)}
      />
      <ColorPickerObject
        visible={picker}
        colorChangeHandler={colorChangeHandler}
      />
    </>
  );
};
