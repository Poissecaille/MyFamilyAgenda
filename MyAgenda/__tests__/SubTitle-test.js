import {SubTitle} from '../old/SubTitle';
import { expect } from '@jest/globals';
import React from 'react'
import {create} from 'react-test-renderer';
// Note: test renderer must be required after react-native.

const tree= create(<SubTitle value="this is a test"/>);

test('test SubTitle component', () => {
    expect(tree.toJSON()).toMatchSnapshot();
    const text = tree.root.findByProps({idTest: "subtitle"}).props
    expect(text.children).toEqual("this is a test");
});