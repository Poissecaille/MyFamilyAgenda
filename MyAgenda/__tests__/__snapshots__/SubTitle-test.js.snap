// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SubTitle component 1`] = `
<Text
  idTest="subtitle"
  style={
    Array [
      Object {
        "color": "#6D8CD3",
        "fontFamily": "PlayfairDisplay-SemiBold",
        "fontSize": 15,
        "textAlign": "left",
      },
    ]
  }
>
  this is a test
</Text>
`;
